impl/.rustfmt.toml

```diff
@@ -0,0 +1 @@
+tab_spaces = 2
```

Comment:
Sorry for this. Honestly this two space thing is a hold over I've had from long long ago back before four spaces was a strong standard. I've intended to make a pass through and fix everything to four spaces, but I've been lazy.

Reply:
I figured it was more a google culture thing, I am agnostic and can convert it to 4 spaces (maybe in a followup PR as it will be a reformat)

I only added in rustfmt because I try to avoid all dogma on formatting, even if I _personally_ dont like some of its rules the fact that its the acceptable standard makes the silly arguments go away.